lib/lockfile.ml

```diff
@@ -131,35 +131,96 @@ module Root_packages = struct
 end
 
 module Depends = struct
-  type t = (string * string) list
+  type t = {
+    dependencies : (OpamPackage.Name.t * OpamPackage.Version.t) list;
```

Comment:
I think this should rather be a list of records, each record being a package and a vendor boolean.

This quite complexified the code compared to what it used to be.

Also note that `OpamPackage.Name.t * OpamPackage,Version.t` is `OpamPackage.t`

Reply:
You're right, a better representation of `vendor` removes a lot of complicated logic.

The `Name.t * Version.t` thing is a good point, I thought there was more to that datatype. I replaced that by using `OpamPackage.t` directly, at which point the `Opam.t` type became just an alias for `OpamPackage.t` with only identity-functions and functions forwarded to `OpamPackage.*` with the only code being `pp` (unused) and `raw_pp` (moved to `Opam.Pp`).